src/realm/dictionary.cpp

```diff
@@ -456,16 +456,11 @@ void Dictionary::sort(std::vector<size_t>& indices, bool ascending) const
     align_indices(indices);
     auto b = indices.begin();
     auto e = indices.end();
-    if (ascending) {
-        std::sort(b, e, [this](size_t i1, size_t i2) {
-            return get_any(i1) < get_any(i2);
-        });
-    }
-    else {
-        std::sort(b, e, [this](size_t i1, size_t i2) {
-            return get_any(i1) > get_any(i2);
-        });
-    }
+    std::sort(b, e, [this](size_t i1, size_t i2) {
+        return get_any(i1) < get_any(i2);
+    });
+    if (!ascending)
+        std::reverse(b, e);
```

Comment:
What's the performance hit in this case?
Compared to having a ternary in the comparison statement?

Reply:
It turns out that sorting Dictionaries is just really slow since every read is a cluster tree lookup and it's not remotely close to the point where it's worth caring about minor inefficiencies.